packages/cascara/src/ui/Table/TechnicalDesign.mdx

```diff
@@ -0,0 +1,138 @@
+---
+name: Spec
+menu: Table
+route: /table-spec
+---
+
+# Technical Design - Espressive Table
+
+# Current implementations
+
+As of `goldengirls` release, we use tables in the following applications:
+
+Barista:
+
+- Admin
+  - Service Department
+  - Service Teams
+  - Barista FAQs
+  - Users
+  - Locations
+
+Caffeine
+
+- Configuration
+- Entities
+- Localization
+- Training
+- Variables
+
+## Features in tables
+
+The features in those tables vary from one implementation to another, but the general idea is to present the data in a tabular way, ussually also providing the means for data persistance. The aggregated features list is the following:
+
+**Item creation**:
+Some tables allow the user to create a new entry in the table. Adding an empty row.
+
+**Inline edition**:
+Some tables allow the user to edit the data, replacing the display representation with some JSX that allows the mutation of the data.
+
+**Item deletion**:
+Some tables allow the user to delete record via a button displayed at the row level.
+
+**Custom actions**:
+In most implementations, tables add a trailing column to each row. This row can display `action buttons`, which can be mapped to corresponding actions over the entire row. i.e. edit, delete, cancel, etc.
+
+## Data representation / mutation
+
+The way each cell represents the data for either display or mutation purposes not only depends on the data type but also on business logic, in most cases, values are easily coerced to string for display purposes and mutations are easily achieved via a controlled input component. The rest of scenarios require a more complex implementation, e.g. the `employee` field in the `Users` table from `Barista Admin` which under the carpet is a single `eid`, but the representation of that cell is an avatar with the user name it points to, although it could be any other representation, specially when editing the value.
```

Comment:
by `mutation` I mean `to edit` the cells in a row and then notify upstream (parent component) about the change. I gues `edit` is a better term.

The header is misleading, it should read `Data representation for display and edit purposes`. The point I am trying to make is that the representation of the data depends on whether the cell is in`display` or `edit` mode.  

> The rest of scenarios require a more complex implementation, e.g. the `employee` field in the `Users` table from `Barista Admin` which under the carpet is a single `eid`, but the representation of that cell is an avatar with the user name it points to

Reply:
Okay. This makes more sense then. 👍🏽 

This brings up another point that I feel this spec might not capture now that you mention this. We should probably capture somewhere a statement that the data does not get edited directly inside of the table Context or State itself. If data is in an edit mode, and a CRUD action fires, the logic for submitting that change to an endpoint and handling all of that is separate, and when the response comes back, the `data` prop gets updated with whatever new data is present. React will do its magic evaluating what parts of the DOM should get updated when this happens.

Reply:
```suggestion
The way each cell represents the data for either display or edit purposes not only depends on the data type but also on business logic, in most cases, values are easily coerced to string for display purposes and updates are easily achieved via a controlled input component. The rest of scenarios require a more complex implementation, e.g. the `employee` field in the `Users` table from `Barista Admin` which under the carpet is a single `eid`, but the representation of that cell is an avatar with the user name it points to, although it could be any other representation, specially when editing the value.

Note: The updates made to any row would be notified upstream via `events` so the parent component is able to persist those changes. The means by which the table enables data mutation are to be provided by the developer in the form of `triggers`, the table doesn't provide such means.
```